kv/kvserver: TestPromoteNonVoterInAddVoter failed

kv/kvserver.TestPromoteNonVoterInAddVoter [failed](https://tanzanite.cluster.engflow.com/invocations/default/160e1fdc-ddf4-478a-b594-7c9844e06fa9?testReportRun=3&testReportShard=36&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.1 @ [557f4e6fe89d683d49082d88ba5adf6660e8f16f](https://github.com/cockroachdb/cockroach/commits/557f4e6fe89d683d49082d88ba5adf6660e8f16f):


```
=== RUN   TestPromoteNonVoterInAddVoter
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestPromoteNonVoterInAddVoter3588943390
    test_log_scope.go:81: use -show-logs to present logs inline
    replicate_queue_test.go:2164: condition failed to evaluate within 45s: from replicate_queue_test.go:2179: Range counts are not all equal
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestPromoteNonVoterInAddVoter3588943390
--- FAIL: TestPromoteNonVoterInAddVoter (96.24s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=36</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPromoteNonVoterInAddVoter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -2118,6 +2118,7 @@ func TestPromoteNonVoterInAddVoter(t *testing.T) {
 	// rebalancing to ensure all stores have the same range count initially, due
 	// to slow heartbeats.
 	skip.UnderStress(t)
+	skip.UnderDeadlock(t)
 	skip.UnderRace(t)
 
 	ctx := context.Background()

